x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java

```diff
@@ -480,6 +480,17 @@ public void upgradeSnapshot(SnapshotUpgradeTask task, Consumer<Exception> closeH
                     );
                     return;
                 }
+                if (resetInProgress) {
+                    closeHandler.accept(
+                        new ElasticsearchStatusException(
+                            "cancelling model upgrade for snapshot [{}] for job [{}] due to feature reset",
+                            RestStatus.NOT_FOUND,
+                            jobId,
+                            snapshotId
+                        )
+                    );
+                    return;
```

Comment:
Like the other case, we should return `null` to the close handler here, and not log anything at all (or log at trace level).  A feature reset means it's totally expected for nothing related to ML to exist afterwards, and anything in progress to have been cancelled without verbose commentary.